src/components/common/TmBalance.vue

```diff
@@ -299,7 +305,16 @@ export default {
     // only be ready to withdraw of the validator rewards are loaded and the user has rewards to withdraw
     // the validator rewards are needed to filter the top 5 validators to withdraw from
     readyToWithdraw() {
-      return this.overview.totalRewards > 0
+      if (this.overview.rewards && this.overview.rewards.length > 0) {
+        const allTotalRewards = this.overview.rewards.map(reward =>
+          this.calculateTotalRewardsDenom(reward.denom)
```

Comment:
are rewards by validator? then `allTotalRewards` would hole multiple records per denom right?

Reply:
yes, rewards are by validators. And yes, allTotalRewards returns an array like this, just with the total amounts per token: `[0.0623, 0.5566, 3.2456 ... ]`

Reply:
it will return multiple records per denom. doesn't really matter but it is badly coded...
you also call `calculateTotalRewardsDenom` which is costly
instead do a computed, that will create one map `aggregatedRewards` with the structure `{ denom: amount }`

Reply:
Yes, I noticed what you said. It returns the same amount per repeated denom in rewards. I guess better to change it for good engineering

Reply:
not suuuuper important as your solution works. but maybe you can take the extra half an hour to practice proper code.

Reply:
xDDDDD

Reply:
OK, I was a long while trying to practice "proper" code, but at the end and after unsuccessfully trying to do it with a `reduce`, I came to the conclusion the best is to just filter the repeated denoms before using `calculateTotalRewardsDenom`. 

I don't know, having this function already in place, I think it is overcomplicating things adding a new computed to do almost the same thing 🤷‍♀

Reply:
please remember me to do this with you in the 1:1

Reply:
OK👌